// Copyright 2015 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package safe_browsing;

message StateStoreData {
  message Incidents {
    // The set of (key,digest) pairs for incidents of a specific type.
    // This is wire-compatible with map<string, uint32> key_to_digest = 1;
    message KeyDigestMapFieldEntry {
      optional string key = 1;
      optional uint32 digest = 2;
    }
    repeated KeyDigestMapFieldEntry key_to_digest = 1;
  }
  // The set of all (key,digest) pairs for all incident types reported.
  // This is wire-compatible with map<int32, Incidents> type_to_incidents = 1;
  message TypeIncidentsMapFieldEntry {
    optional int32 type = 1;
    optional Incidents incidents = 2;
  }
  repeated TypeIncidentsMapFieldEntry type_to_incidents = 1;
}
